src/engine/Minus.cpp

```diff
@@ -96,18 +96,24 @@ size_t Minus::getCostEstimate() {
 }
 
 // _____________________________________________________________________________
-auto Minus::makeUndefRangesChecker(bool left) const {
+auto Minus::makeUndefRangesChecker(bool left, const IdTable& idTable) const {
   std::variant<ad_utility::Noop, ad_utility::FindSmallerUndefRanges> findUndef;
```

Comment:
As you have an explicit return type here which is easily nameable, 
then you can give it a name via `using` in the header and make it the return type of this function.
This increases the readability of your docstring automatically.